### PR TITLE
Consolidate RelationalTypeMapping.Clone methods into one…

### PR DESCRIPTION
**Pull request by AndriySvyryd**
… and other type mapping refactorings.

### CODE REVIEW

File: src/EFCore.Relational/breakingchanges.netcore.json [hunk 1]
```diff
@@ -113,5 +113,100 @@
       "TypeId": "public interface Microsoft.EntityFrameworkCore.Query.IQueryMethodProvider",
       "MemberId": "System.Reflection.MethodInfo get_FastQueryMethod()",
       "Kind": "Addition"
+    },    
+    {
+      "TypeId": "public class Microsoft.EntityFrameworkCore.Storage.BoolTypeMapping : Microsoft.EntityFrameworkCore.Storage.RelationalTypeMapping",
+      "MemberId": "public override Microsoft.EntityFrameworkCore.Storage.RelationalTypeMapping Clone(System.String storeType, System.Nullable<System.Int32> size)",
```
Comment (ajcvickers) [inline src/EFCore.Relational/breakingchanges.netcore.json]: I've convinced myself that this isn't a compile break. Are we sure that it isn't a binary break?

Comment (AndriySvyryd) [reply src/EFCore.Relational/breakingchanges.netcore.json]: The base method is still there and nobody should be calling the override directly.

File: test/EFCore.Design.Tests/Scaffolding/Internal/ScaffoldingTypeMapperSqlServerTest.cs [hunk 2]
```diff
@@ -78,14 +80,6 @@ public void Maps_normal_varbinary_sized_column()
             AssertMapping<byte[]>(mapping, inferred: true, maxLength: 200, unicode: null, fixedLength: null);
         }
 
-        [Fact]
-        public void Maps_normal_binary_max_column()
```
Comment (ajcvickers) [inline test/EFCore.Design.Tests/Scaffolding/Internal/ScaffoldingTypeMapperSqlServerTest.cs]: Why are tests removed here?

Comment (AndriySvyryd) [reply test/EFCore.Design.Tests/Scaffolding/Internal/ScaffoldingTypeMapperSqlServerTest.cs]: These types are invalid

Comment (ajcvickers) [reply test/EFCore.Design.Tests/Scaffolding/Internal/ScaffoldingTypeMapperSqlServerTest.cs]: Interesting. Didn't know that.

**Pull request merged**